using System;
using System.Threading.Tasks;
using BCEventsApp.Database;
using BCEventsApp.Exceptions;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTest.Database
{
    [TestClass]
    public class UserActionsTest
    {
////        [TestMethod]
////        public async Task TestGetAllCalendars()
////        {
////            var idList = await UserActions.GetAllCalendars("testa");
////        }
////
//        [TestMethod]
//        public async Task TestUserGetAllSubscribedEvents()
//        {
//            var idList = await UserActions.GetAllSubscribedEvents("testb");
//        }
////
////        [TestMethod]
////        public async Task TestSubscribeCalendar()
////        {
////            try
////            {
////                await UserActions.SubscribeCalendar("testa", 34);
////            }
////            catch (CustomException e)
////            {
////                var msg = e.GetMessage();
////            }
////        }
////
////        [TestMethod]
////        public async Task TestUnsubscribeCalendar()
////        {
////            try
////            {
////                await UserActions.UnsubscribeCalendar("testa", 34);
////            }
////            catch (CustomException e)
////            {
////                var msg = e.GetMessage();
////            }
////        }
////
////        [TestMethod]
////        public async Task TestSubscribeEvent()
////        {
////            try
////            {
////                await UserActions.SubscribeEvent("testb", 33);
////            }
////            catch (CustomException e)
////            {
////                var msg = e.GetMessage();
////            }
////        }
////
////        [TestMethod]
////        public async Task TestUnsubscribeEvent()
////        {
////            try
////            {
////                await UserActions.UnsubscribeEvent("testb", 33);
////            }
////            catch (CustomException e)
////            {
////                var msg = e.GetMessage();
////            }
////        }
////
////        [TestMethod]
////        public async Task TestAttendEvent()
////        {
////            try
////            {
////                await UserActions.AttendEvent("testb", 33);
////            }
////            catch (CustomException e)
////            {
////                var msg = e.GetMessage();
////            }
////        }
////
////        [TestMethod]
////        public async Task TestWithdrawEvent()
////        {
////            
////            await UserActions.WithdrawEvent("testb", 33);
////        }
////
////        [TestMethod]
////        public async Task TestCreateCalendar()
////        {
////            await UserActions.CreateCalendar("testb", "Testb Calendar 4");
////        }
//
////        [TestMethod]
////        public async Task TestCreateEvent()
////        {
////            var start = new DateTime(2016, 1, 1, 14, 30, 0 );
////            var end = new DateTime();
////            var rRule = "FREQ=WEEKLY;BYDAY=TU,TH;DTSTART=20160101T203000Z;UNTIL=20161231T060000Z";          
////            var eid = await UserActions.CreateEvent("testb", "Repeat2", "Back and forth", "there", start, 45, end, rRule, 28);
////        }
//
////        [TestMethod]
////        public async Task TestGetPrimaryCalendarId()
////        {
////            var id = await UserActions.GetPrimaryCalendarId("testb");
////        }
//
////        [TestMethod]
////        public async Task TestDeleteEvent()
////        {
////            await UserActions.DeleteEvent(68);
////        }
    }
}
      
 
